### PR TITLE
stderr output to console, OpenAI quirks

### PR DESCRIPTION
**Pull request by swg**
This pull request starts dumping stderr to console
It also fixes some quirks for compatibility with the OpenAI API (newlines, defaulting to infinity if max_tokens is unset)

### CODE REVIEW

File: defaults.js [hunk 1]
```diff
@@ -25,7 +25,11 @@ export const getArgs = (args) => {
 	const convertedArgs = {};
 	Object.keys(args).forEach((a) => {
 		if (!!openAiToLlamaMapping[a]) {
-			convertedArgs[openAiToLlamaMapping[a]] = args[a];
+			if (a === 'max_tokens' && args[a] === null) {
+				convertedArgs[openAiToLlamaMapping[a]] = -1;
+			} else {
+				convertedArgs[openAiToLlamaMapping[a]] = args[a];
+			}
```
Comment (keldenl) [inline defaults.js:32]: if we want the default to be -1 we should just keep the original code the same here and update the defaultParams for --n_predict to be -1.

File: routes/chatRoutes.js [hunk 2]
```diff
@@ -162,8 +162,36 @@ assistant:`;
 	console.log(`\n=====  REQUEST  =====\n${messagesToString([lastMessage])}`);
 
 	let stdoutStream = global.childProcess.stdout;
+	let stderrStream = global.childProcess.stderr;
 
-	const readable = new ReadableStream({
+	const stderr = new ReadableStream({
```
Comment (keldenl) [inline routes/chatRoutes.js:167]: what's the difference between stdout and steer? sorry not super well versed in this stuff lol

Comment (swg) [reply routes/chatRoutes.js:167]: stdout, stdin and stderr are the three text streams setup by the OS, available to any program to communicate with the terminal/console. llama just uses stderr to output a bunch of useful things, including errors, and we can potentially parse this for errors to return to clients for each invocation, although it's not used for such yet.

File: defaults.js (hunk 1 above)
Comment (swg) [reply defaults.js:32]: I don't have time this morning to look at all of the code, but the behavior of the OpenAI API is to use the default (in this case -1 for ChatCompletion) if a null is explicitly passed. So there's some trickiness in preserving that quirk. It probably applies to every other option passed via the API that has a default.

In the client that I was attempting to fix, unrelated to this project, I came to the realization last night that it was incorrectly using the max_token to determine what is actually in the context (to limit the number of messages sent on each ChatCompletion) versus using a lookup table depending on the model (different models have different context limits.) So in the client I'm talking about, again unrelated to this project, there's more fixes to be made.

I'll rebase and fix according to your feedback as soon as I get a chance if you are interested in preserving this quirk of the OpenAI API, in your project. Thank you!

Comment (keldenl) [reply defaults.js:32]: sounds good! i'll be on the lookout!

**Review by keldenl — APPROVED** [review]
I went ahead and fixed the merge conflicts, going to merge this!

**Pull request merged**